frontend/src/modules/otherlocales/components/Locales.test.js

```diff
@@ -17,18 +17,54 @@ describe('<Locales>', () => {
             locale: 'kg',
             project: 'tmo',
         };
+        const user = {}
         const wrapper = shallow(
-            <Locales otherlocales={ otherlocales } parameters={ params } />
+            <Locales
+                otherlocales={ otherlocales }
+                parameters={ params }
+                user={ user }
+            />
         );
 
         expect(wrapper.find('Translation')).toHaveLength(3);
     });
 
+    it('shows preferred locales first', () => {
+        const otherlocales = {
+            translations: [
+                { code: 'ar' },
+                { code: 'br' },
+                { code: 'cr' },
+            ],
+        };
+        const params = {
+            locale: 'kg',
+            project: 'tmo',
+        };
+        const user = {
+            isAuthenticated: true,
+            preferredLocales: ['ab', 'br', 'cd']
+        };
+        const wrapper = shallow(
+            <Locales
+                otherlocales={ otherlocales }
+                parameters={ params }
+                user={ user }
+            />
+        );
+
+        expect(wrapper.find('Translation:first-child .stress').text()).toContain('br');
```

Comment:
This test fails with:
```
Method “text” is only meant to be run on a single node. 0 found instead."
```

`Translation:first-child` is found, but for some reason `Translation:first-child .stress` isn't.

Reply:
That is because `shallow` only creates the top component, and none of its children. In this case, that means `Translation` is not constructed, and thus you cannot access its content. But that is good, because in tests you should only test the component you're interested in, and no others.

In this case I believe you want to test the outcome of `sortByPreferred` instead of looking at what's rendered. You can do that by calling `wrapper.instance().sortByPreferred()`.

Reply:
Note that moving the ordering logic to a selector will make it a lot simpler to test, which I believe is also a good argument in favor of the selector. :)